Makefile

```diff
@@ -305,8 +305,10 @@ test.conformance: go-junit-report
 		-race $(GOTESTFLAGS) \
 		-timeout $(INTEGRATION_TEST_TIMEOUT) \
 		-parallel $(NCPU) \
-		./test/conformance |
-	$(GOJUNIT) -iocopy -out $(JUNIT_REPORT) -set-exit-code --parser gotest
+		./test/conformance > conformance-tests.out
+	exit_code=$$?
+	$(GOJUNIT) -in conformance-tests.out -out $(JUNIT_REPORT) -parser gotest
+	exit $$exit_code
```

Comment:
Have you tried running this just without the pipe? Can it set the proper exit code when it reads from the file instead of a pipe?

This way we can do away with explicitly setting the exit code

Reply:
Hmm, looks like indeed it manages to set the exit code once fed from a file 🤷

Reply:
Do we have 
```
set -o pipefail
```
set for bash?

I don't see this anywhere in `Makefile`

Reply:
Having `.SHELLFLAGS := -eu -o pipefail -c` in the Makefile the integration tests target failure was still giving always a 0 exit code.

Reply:
One thing to note is that when the tests fail like this we won't get the report because without the `-` prefix when invoking a command - https://www.gnu.org/software/make/manual/html_node/Errors.html - the error in `go test` will prevent `go-junit` from running.

Which should be fine IMHO because what's point of the report when the tests  failed 🤷 ?

Reply:
You're right it makes sense to track those, at least for the purposes of flakiness tracking through buildpulse. In that case: will `$(GOJUNIT) -in conformance-tests.out -out $(JUNIT_REPORT) -parser gotest` fire when `go test` fails?

Reply:
You're right, [ttps://www.gnu.org/software/make/manual/html_node/Errors.html](https://www.gnu.org/software/make/manual/html_node/Errors.html) will prevent it from running. 😬

Reply:
Okay, it turns out setting `.SHELLFLAGS := -eu -o pipefail -c` in the Makefile works as expected on GH runners ([CI run](https://github.com/Kong/kubernetes-ingress-controller/actions/runs/5091603544/jobs/9151854177)). I couldn't make it work locally though - my MacOS version of Make completely ignores this. 😕 

Anyways, I guess it's fine to merge it as it fixes the problem on CI. @pmalek @programmer04 PTAL

Reply:
It's `make` nobody understands how, where and what happens there :trollface: